clar2wasm/src/standard/standard.wat

```diff
@@ -741,6 +741,88 @@
         (call $sqrti-uint (local.get $lo) (local.get $hi))
     )
 
+    (func $bit-and (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+        (i64.and (local.get $a_lo) (local.get $b_lo))
+        (i64.and (local.get $a_hi) (local.get $b_hi))
+    )
+
+    (func $bit-not (type 3) (param $a_lo i64) (param $a_hi i64) (result i64 i64)
+          ;; wasm does not have bitwise negation, but xoring with -1 is equivalent
+          (i64.xor (local.get $a_lo) (i64.const -1))
+          (i64.xor (local.get $a_hi) (i64.const -1))
+    )
+
+    (func $bit-or (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+          (i64.or (local.get $a_lo) (local.get $b_lo))
+          (i64.or (local.get $a_hi) (local.get $b_hi))
+    )
+
+    (func $bit-xor (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+          (i64.xor (local.get $a_lo) (local.get $b_lo))
+          (i64.xor (local.get $a_hi) (local.get $b_hi))
+    )
+
+    (func $bit-shift-left (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
+          ;; only b_lo is useful here since we will take the reminder by 128
+          ;; n % 128 == n & 127 == n & 0x7f
+          (local.set $b_lo (i64.and (local.get $b_lo) (i64.const 0x7f)))
+          ;; Two cases when we shift:
+          ;; (1) we shift by a 0 <= shift < 64: we have to split the lower bits into the carried bits and the rest, then we shift
+          ;;     the rest, we shift the higher part and we add the carry to the higher part.
+          ;; (2) we shift by a 64 <= shift < 128: lower bits are automatically 0, and higher bits are the lower bits shifted by (shift - 64).
+          (if (result i64 i64) (i64.lt_u (local.get $b_lo) (i64.const 64))
+              (then ;; (1)
+               (local.set $b_hi ;; using $b_hi for storing overflow bits
+                          (select ;; that's a hack to workaround wasm shift by 64 has no effect
+                           (i64.const 0)
+                           (i64.shr_u (local.get $a_lo) (i64.sub (i64.const 64) (local.get $b_lo)))
+                           (i64.eqz (local.get $b_lo))))
+               (i64.shl (local.get $a_lo) (local.get $b_lo)) ;; lower_bits <<= shift
+               (i64.or (i64.shl (local.get $a_hi) (local.get $b_lo))
+                       (local.get $b_hi))) ;; higher_bits = (higher_bits << shift) | carry
+              (else ;; (2)
+               (i64.const 0)
+               (i64.shl (local.get $a_lo) (i64.sub (local.get $b_lo) (i64.const 64))))))
+
+    (func $bit-shift-right-uint (type 1) (param $a_lo i64) (param $a_hi i64) (param $b_lo i64) (param $b_hi i64) (result i64 i64)
```

Comment:
Since this function is not explained, maybe we should add a comment to say that it is a similar reasoning as shift left? (and same for shift right int)

Reply:
Most functions are not explained, i think we don't need to overdo the docs here, it's easy to see that it's similar

Reply:
Except for functions in one statement that are obvious, every functions have comments.

Reply:
I added some comments